sofana/R3BSofFrsAnalysis.cxx

```diff
@@ -302,12 +304,23 @@ void R3BSofFrsAnalysis::Exec(Option_t* option)
             i_s8cave = i;
         }
         if (isnan(tof) || tof < 0)
+        {
             continue;
-        beta[i] = fPathLength[i] / (tof + fTofOffset[i]); // ToFCalib
+        }
+        else
+        {
+            // beta[i] = fPathLength[i] / (tof + fTofOffset[i]); // ToFCalib
+            beta.push_back(fPathLength[i] / (tof + fTofOffset[i]));
+        }
     }
     // Velocity correlation conditions
-    if (fBetaCorr && beta[i_s2cave] > 0. && beta[i_s8cave] > 0. && TMath::Abs(beta[i_s2cave] - beta[i_s8cave] < 0.01))
-        beta[fNbTof] = beta[i_s2cave];
+    // Note that TMath::Abs(beta.at(i_s2cave) - beta.at(i_s8cave) < 0.01) should be TMath::Abs(beta.at(i_s2cave) -
+    // beta.at(i_s8cave)) < 0.01
+    if (fBetaCorr && beta.at(i_s2cave) > 0. && beta.at(i_s8cave) > 0. &&
+        TMath::Abs(beta.at(i_s2cave) - beta.at(i_s8cave) < 0.01))
+    {
+        beta.at(fNbTof) = beta.at(i_s2cave);
```

Comment:
In case we change line 285 to `std::vector<Double_t> beta; `, then change here to `beta.push_back(beta.at(i_s2cave));` and change the line 326 as well.